mirrord/agent/src/main.rs

```diff
@@ -516,7 +549,9 @@ async fn start_agent() -> Result<()> {
 
     let dns_api = DnsApi::new(state.container_pid(), 1000);
 
-    let (sniffer_task, sniffer_status) = {
+    let (sniffer_task, sniffer_status) = if matches!(args.mode, cli::Mode::Targetless) {
```

Comment:
nitpick, optional: we can consider having an `args.mode.is_targetless()` function for these checks.